server/api/externalapi.ts

```diff
@@ -325,13 +337,15 @@ class ExternalAPI {
 
   private serializeCacheKey(
     endpoint: string,
-    params?: Record<string, unknown>
+    params?: Record<string, unknown>,
+    headers?: Record<string, unknown>
   ) {
-    if (!params) {
-      return `${this.baseUrl}${endpoint}`;
+    const key = `${this.baseUrl}${endpoint}`;
+    if (!params && !headers) {
+      return key;
     }
 
-    return `${this.baseUrl}${endpoint}${JSON.stringify(params)}`;
+    return `${key}${JSON.stringify({ params, headers })}`;
   }
 
   private async getDataFromResponse(response: Response) {
```

Comment:
IMO the previous implementation was good, but the 2nd parameter of the function should probably not be called "params", but something like "data", as this parameter can contains URL params, headers or data (see the missing "data" you forgot above)